pymathics/graph/base.py

```diff
@@ -1621,25 +1622,66 @@ def eval_1(self, vertices, edges, evaluation, options):
         )
 
 
-class GraphBox(GraphicsBox):
-    def _graphics_box(self, elements, options):
-        evaluation = options["evaluation"]
-        graph, form = elements
-        primitives = graph._layout(evaluation)
-        graphics = Expression(SymbolGraphics, primitives)
-        graphics_box = Expression(SymbolMakeBoxes, graphics, form).evaluate(evaluation)
-        return graphics_box
+class GraphBox(BoxElementMixin):
+    def __init__(self, G, **options):
+        self.G = G
+        self.options = options
 
-    def boxes_to_text(self, elements, **options):
-        return "-Graph-"
+    def boxes_to_b64text(
+        self, elements: Tuple[BaseElement] = None, **options
+    ) -> Tuple[bytes, Tuple[int, int]]:
+        """
+        Produces a base64 png representation and a tuple with the size of the pillow image
+        associated to the object.
+        """
+        contents = self.boxes_to_png(elements, **options)
+        encoded = base64.b64encode(contents)
+        encoded = b"data:image/png;base64," + encoded
+        return encoded
+
+    def boxes_to_png(self, elements=None, **options) -> Tuple[bytes, Tuple[int, int]]:
+        """
+        returns a tuple with the set of bytes with a png representation of the image
+        and the scaled size.
+        """
+        return png_format_graph(self.G, **self.options)
 
-    def boxes_to_xml(self, elements, **options):
-        # Figure out what to do here.
-        return "-Graph-XML-"
+    def boxes_to_svg(self, elements=None, **options):
+        return svg_format_graph(self.G, **self.options)
+
+    def boxes_to_tex(self, elements=None, **options) -> str:
+        """
+        Store the associated image as a png file and return
+        a LaTeX command for including it.
+        """
+
+        data = self.boxes_to_png(elements, **options)
+        size = (
+            800,
+            800,
```

Comment:
In printed docs 800, 800 seems to stretch things out along the Y axis. Emperically 800 x 600 seems to look right.